### PR TITLE
HPCC-20669 Label for checkbox not display

### PR DESCRIPTION
**Pull request by gaojx**
## Type of change:
- [x ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

### CODE REVIEW

**Comment by gaojx** [comment]
@timothyklemm Please review. Easy change to fix UI issue.

**Comment by timothyklemm** [comment]
The commit message format is improper. See https://github.com/hpcc-systems/HPCC-Platform/wiki/Git-commit-message-guidelines, The format is important for the automation to correctly update the JIRA ticket.

**Review by timothyklemm — COMMENTED** [review]
Have you handled all occurrences of the problem? I see two locations in batchworkunits.xslt that seem to do the same thing that you're changing here (lines 856 and 862). That is, they are treating 'input' as a non-empty element with text content.

**Comment by gaojx** [comment]
Good point. I only checked all files in esp/xslt directory. It seems that there are quite some use in <input> in esp/eclwatch. Will look at them.

**Comment by gaojx** [comment]
Add some in esp/services too

**Comment by richardkchapman** [comment]
Correct format for commit title is:

HPCC-20669 Label for checkbox not displayed

**Comment by richardkchapman** [comment]
@timothyklemm Please re-review

File: esp/services/ws_machine/metrics.xslt [hunk 1]
```diff
@@ -520,11 +520,13 @@
                <table>
                   <tr>
                      <th id="selectAll1" colspan="5">
-                       <input type="checkbox" id="SelectAllCheckBox" title="{$hpccStrings/st[@id='SelectDeselectAll']}" onclick="selectAll0(this.checked)">
+                       <label><input type="checkbox" id="SelectAllCheckBox" title="{$hpccStrings/st[@id='SelectDeselectAll']}" onclick="selectAll0(this.checked)">
                          <xsl:if test="$selectAllChecked &gt; 0">
                            <xsl:attribute name="checked">true</xsl:attribute>
                          </xsl:if>
-                       <xsl:value-of select="$hpccStrings/st[@id='SelectAllOrNone']"/></input>
+                        </input>
```
Comment (timothyklemm) [inline esp/services/ws_machine/metrics.xslt]: Should the input element be empty?

File: esp/eclwatch/ws_XSLT/batchworkunits.xslt [hunk 2]
```diff
@@ -378,7 +378,7 @@
                             <td><b>Batch Workunits</b>
                             </td>
                             <td>
-                                <input type="checkbox"  id="AutoRefresh" value="Auto Refresh" onclick="setAutoRefresh(this);"/> Auto Refresh
+                                <label><input type="checkbox"  id="AutoRefresh" value="Auto Refresh" onclick="setAutoRefresh(this);"/>Auto Refresh</label>
```
Comment (timothyklemm) [inline esp/eclwatch/ws_XSLT/batchworkunits.xslt]: Is it appropriate to remove the space? Most inputs don't include spaces, but some include non-breaking spaces and a couple include regular spaces.

Comment (gaojx) [reply esp/eclwatch/ws_XSLT/batchworkunits.xslt]: Yes, it is empty. This is another way to label a checkbox:
   <label><input type="checkbox" ... /> Label Text </label>

File: esp/services/ws_machine/metrics.xslt (hunk 1 above)
Comment (gaojx) [reply esp/services/ws_machine/metrics.xslt]: Yes, it is empty. This is another way to label a checkbox:
 `<label><input type="checkbox" ... /> Label Text </label>`

Comment (timothyklemm) [reply esp/services/ws_machine/metrics.xslt]: My bad. The transform content threw me off - I didn't realize it was modifying the attribute list.

**Comment by timothyklemm** [comment]
Changes look good. Just need to squash commits and edit the commit message.

**Comment by gaojx** [comment]
Commits squashed and commit message changed to meet standard.

**Pull request merged**